xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp

```diff
@@ -628,6 +628,7 @@ bool CVideoPlayerVideo::ProcessDecoderOutput(int &decoderState, double &frametim
         msg.player = VideoPlayer_VIDEO;
         msg.cachetime = DVD_MSEC_TO_TIME(50); // TODO
         msg.cachetotal = DVD_MSEC_TO_TIME(100); // TODO
+        msg.next_pts = this->m_messageQueue.GetTimeBack();
```

Comment:
It IS hacky :-) but isn't the messagequeue the last buffer before decoding?
As far as I understood these messages will be - poped from queue - directly decoded and displayed.

I haven't seen a better PTS for continuing after a stream change